src/test/kotlin/sirius/kernel/commons/StringsTest.kt

```diff
@@ -319,4 +319,25 @@ class StringsTest {
         assertEquals("A", Strings.limit("A", 1, false))
     }
 
+    @Test
+    fun truncateMiddle() {
+        assertEquals("", Strings.truncateMiddle(null, 4, 4))
+        assertEquals("", Strings.truncateMiddle(null, -4, 4))
+        assertEquals("", Strings.truncateMiddle(null, 4, -4))
+        assertEquals("", Strings.truncateMiddle("", 4, 4))
+        assertEquals("", Strings.truncateMiddle("", -4, 4))
+        assertEquals("", Strings.truncateMiddle("", 4, -4))
+        assertEquals("", Strings.truncateMiddle("1234-6789", -4, 4))
+        assertEquals("", Strings.truncateMiddle("1234-6789", 4, -4))
+        assertEquals("", Strings.truncateMiddle("1234-6789", -4, -4))
+        assertEquals("", Strings.truncateMiddle("1234-6789", 0, 0))
+        assertEquals("1234-6789", Strings.truncateMiddle("1234-6789", 40, 0))
+        assertEquals("1234-6789", Strings.truncateMiddle("1234-6789", 0, 40))
+        assertEquals("1234-6789", Strings.truncateMiddle("1234-6789", 5, 4))
+        assertEquals("1234-6789", Strings.truncateMiddle("1234-6789", 4, 5))
+        assertEquals("1234…[truncated]…6789", Strings.truncateMiddle("1234-6789", 4, 4))
```

Comment:
**Question:** Here the result gets longer as the input again?? whats the usecase of this method if it makes a text longer in some cases?

My suggestion: Always return the original String when (preserveStart + signal.length + preserveEnd) is greater or equal to the input.length

Reply:
*Answer:* 
- the use-case is to truncate Strings with a signal showing that something has been truncated. For example: lets say I dont care for the first part of the String, which is why I set `charsToPreserveAtStart` to `0` and the other parameter to `15` because I only want to see the last 15 chars... Then it would be helpful to see that something got `truncated`. Since I don't know how long the input will be, at least my output will be "stable" as its always signal + `charsToPreserveAtEnd` in this case.

But I can change it accordingly if you like

Reply:
But when I want to truncate a string its because I want to restrict it to a certain length, so making it longer by "truncating" it defeats the purpose in my opinion.

> Then it would be helpful to see that something got `truncated`
Thats not what I am critiquing, I am critiquing that we truncate when we don't gain a benefit from it, but instead a negative (as in lost information and sometimes even a longer string than before)

Reply:
IMHO: Changing it to your suggestion makes the method less intuitive because then I need to know that the string has to be a minimum length of 14 chars before truncating takes place, also I need to make sure that the sum of `charsToPreserveFromStart`, `charsToPreserveFromEnd`, `13` makes sense, otherwise I get the input string back

Reply:
Discussion continued in google-meet...